src/test/java/org/broadinstitute/hellbender/utils/config/ConfigIntegrationTest.java

```diff
@@ -0,0 +1,126 @@
+//      A better integration test, I think, would be to create a private static dummy tool
+//      class here that just dumps selected items from its configuration to a file in apply(), run that dummy tool
+//      in your test, and assert that the configuration dumped is consistent with the one specified on the command line.
+//
+//      You could have two test cases: one with the default config file, and one with a user-specified config file
+//      that alters one or more default settings.
+
+        final File tmpFile = getSafeNonExistentFile("ConfigIntegrationTest.output.txt");
+        tmpFile.deleteOnExit();
+
+        // Create some arguments for our command:
+        final ArgumentsBuilder args = new ArgumentsBuilder();
+        args.add("--" + StandardArgumentDefinitions.GATK_CONFIG_FILE_OPTION);
+        args.add(configFilePath.toUri().toString());
+        args.add("-" + StandardArgumentDefinitions.OUTPUT_SHORT_NAME);
+        args.add(tmpFile);
+
+        // Run our command:
+        runCommandLine(args.getArgsArray());
+
+        // Now we get to read in the file's contents and compare them to our config settings:
+        final Properties properties = new Properties();
+        try ( final FileInputStream inputStream = new FileInputStream(tmpFile) ) {
+            properties.load(inputStream);
+        }
+        catch (final Exception ex) {
+            throw new GATKException("Test error!", ex);
+        }
+
+        // Create a config object to compare with the properties:
+        final GATKConfig config = ConfigFactory.getInstance().getGATKConfig();
+
+        final LinkedHashMap<String, Object> configMap = ConfigFactory.getConfigMap(config, false);
+
+        // Need to use reflection here to make this test future-proof.
+        // Now we cycle through our interface methods, resolve parameter names,
+        // and assert their equality with the properties we have read:
+        for ( final Class<?> classInterface : ClassUtils.getClassesOfType(Config.class, Arrays.asList(config.getClass().getInterfaces())) ) {
+            for ( final Method propertyMethod : classInterface.getDeclaredMethods() ) {
+
+                // Get the property name:
+                String propertyName = propertyMethod.getName();
+
+                // Get the real property name if we've overwritten it with a key:
+                final Config.Key key = propertyMethod.getAnnotation(Config.Key.class);
+                if ( key != null ) {
+                    propertyName = key.value();
+                }
+
+                try {
+                    Assert.assertEquals(configMap.get(propertyName), propertyMethod.invoke(config));
+                }
+                catch ( final Exception ex ) {
+                    throw new GATKException("Error validating config when reading from config object: " +
+                            config.getClass().getSimpleName() + "::" + propertyMethod.getName(), ex);
+                }
+            }
+        }
```

Comment:
If you take my suggestion above and add in a second test case with an override config file, you may need some code here at the end of this test that restores the default GATK config settings.

Reply:
Yeah - this was why I hadn't added in that second test yet.  I was still coming up with a way to do this well.